src/app/state/store.tsx

```diff
@@ -18,7 +18,8 @@ export const middleware: Middleware[] = [
     notificationCheckerMiddleware
 ];
 const defaultMiddlewareOptions = {
-    serializableCheck: process.env.NODE_ENV !== 'test'
+    serializableCheck: process.env.NODE_ENV !== 'test',
+    immutableCheck: false,
```

Comment:
What does this property do? How come this is needed? My main worry here is that we're changing the store properties for the sake of making the tests pass, which can be dangerous, as this affects the whole application, not just the tests.

Reply:
This checks for state mutations I believe, however with it turned on it is currently giving this console warning: 

"ImmutableStateInvariantMiddleware took 106ms, which is more than the warning threshold of 32ms. 
      If your state or actions are very large, you may want to disable the middleware as it might cause too much of a slowdown in development mode."

It's not used in production, only dev

Reply:
Thanks for the explanation In that case, I would revert this change, because all it's doing is telling you that it's taking too long. By setting that property to false, you're not fixing the potential problem, just silencing it. It would be better to leave the immutability check turned on and create a low priority ticket to look into why this thing is taking so long.

If you leave the immutability check to false, then (if I understand it correctly) you would be losing the protection that it gives you (the immutability check throws an error if you change the state outside the store, I think).

Reply:
I've re-enabled and created a ticket for the warning (it's intermittent currently)